pkg/reconciler/v1alpha1/taskrun/resources/output_resource.go

```diff
@@ -105,27 +105,34 @@ func AddOutputResources(
 			}
 		}
 
-		// copy to pvc if pvc is present
-		if pipelineRunpvcName != "" {
-			var newSteps []corev1.Container
-			for _, dPath := range boundResource.Paths {
-				newSteps = append(newSteps, []corev1.Container{{
-					Name:  fmt.Sprintf("source-mkdir-%s", resource.GetName()),
-					Image: *bashNoopImage,
-					Args: []string{
-						"-args", strings.Join([]string{"mkdir", "-p", dPath}, " "),
-					},
-					VolumeMounts: []corev1.VolumeMount{getPvcMount(pipelineRunpvcName)},
-				}, {
-					Name:  fmt.Sprintf("source-copy-%s", resource.GetName()),
-					Image: *bashNoopImage,
-					Args: []string{
-						"-args", strings.Join([]string{"cp", "-r", fmt.Sprintf("%s/.", sourcePath), dPath}, " "),
-					},
-					VolumeMounts: []corev1.VolumeMount{getPvcMount(pipelineRunpvcName)},
-				}}...)
-			}
-			b.Spec.Steps = append(b.Spec.Steps, newSteps...)
+		// Workaround for issue #401. Unless all resource types are implemented as
+		// outputs, or until we have metadata on the resource that declares whether
+		// the output should be copied to the PVC, only copy git and storage output
+		// resources.
+		if (resource.Spec.Type == v1alpha1.PipelineResourceTypeStorage ||
```

Comment:
Suggestion: 
Nested `if` condition is long and hard to follow. I would recommend adding a function like `allowedOutputResources(string) bool` which will validate type `Storage` and `Git` and return boolean. So you can update `if` condition to be like 

```
// copy resources if its supported list and PVC is present
if allowedOutputResource(resource.Spec.Type) && pipelineRunpvcName != "" {
...
}
```

This makes the code more readable.

Reply:
ok

Reply:
Done in an extra commit